go/enclave/rpc/GetBalance.go

```diff
@@ -0,0 +1,44 @@
+package rpc
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ten-protocol/go-ten/go/common"
+	"github.com/ten-protocol/go-ten/go/common/gethencoding"
+	"github.com/ten-protocol/go-ten/go/responses"
+)
+
+func (rpc *EncryptionManager) GetBalance(encryptedParams common.EncryptedParamsGetBalance) (*responses.Balance, common.SystemError) {
+	return withVKEncryption1[hexutil.Big, hexutil.Big](
+		rpc,
+		rpc.config.ObscuroChainID,
+		encryptedParams,
+		// extract sender and arguments
+		func(reqParams []any) (*UserRPCRequest1[hexutil.Big], error) {
+			// Parameters are [Address, BlockNumber]
+			if len(reqParams) != 2 {
+				return nil, fmt.Errorf("unexpected number of parameters")
+			}
+			requestedAddress, err := gethencoding.ExtractAddress(reqParams[0])
+			if err != nil {
+				return nil, fmt.Errorf("unable to extract requested address - %w", err)
+			}
+
+			blockNumber, err := gethencoding.ExtractBlockNumber(reqParams[1])
+			if err != nil {
+				return nil, fmt.Errorf("unable to extract requested block number - %w", err)
+			}
+
+			encryptAddress, balance, err := rpc.chain.GetBalance(*requestedAddress, blockNumber)
```

Comment:
I think the execution has slipped into the parameters function here, and then you pass the balance result into the next part as if it was a param.

Maybe we're thinking about something wrong if the distinction between these two functions isn't that important?

Reply:
Sometimes, in order to extract the sender, you need to execute.
And since the call is already executed, the "Execute" becomes trivial.

In a world where performance wouldn't matter, you'd execute whatever you need to get the sender in the first call. And then re-execute in the second for the actual result